beancount_ethereum/importer.py

```diff
@@ -84,19 +84,49 @@ def _create_posting(
                     # Do not create posting
                     account = None
                 else:
-                    account = f'{self.account_map[address]}:{self.account_suffix(currency)}'
+                    account = (
+                        f"{self.account_map[address]}:{self.account_suffix(currency)}"
+                    )
                 payee = None
         if account:
             posting = Posting(
                 account,
                 Amount(value, self.commodity(currency)),
-                None, None, None, None,
+                None,
+                None,
+                None,
+                None,
             )
         else:
             posting = None
         return posting, payee
 
-    def extract(self, file, existing_entries=None) -> list:
+    def extract_balances(self, file) -> list:
+
+        with open(file.name, 'r') as _file:
+            balances = json.load(_file)
+        entries = []
+
+        for record in balances:
+            meta = new_metadata(file.name, 0)
+            balance_date = datetime.datetime.fromtimestamp(record["time"])
+            currency = record['currency']
+            account = f"{self.account_map[record['address']]}:{self.account_suffix(currency)}"
```

Comment:
> Do you plan to fetch token balances as well?

Yes.
However it is more difficult as there is no method in the API to get all balances of all token for one adress with one call.
You have to call the API and to make one call for each token contract and each adress.
This means that i will have to add a dictionary in the json file to list the token for which we want to control the balance and the corresponding contract adress.

Reply:
> It gives me `KeyError` for `self.accont_map` when I try to run it.
> 
> Probably because the address in `get_base_currency_balances` should be normalized (converted to lower case).

This is strange i have not met this issue. Do you have uppercase letters in your address ?
Could you try with the latest commit ?

Reply:
Yes, my address is checksummed with mixed case.
The latest commit fixed that, thanks